setup.go

```diff
@@ -0,0 +1,815 @@
+}
+
+// ------------------------------ Setup constants ------------------------------
+const ( // state enum
+	stateDeviceType = iota
+	stateDeviceTypeConfirm
+	stateHostedMender
+	stateDemoMode
+	stateServerURL
+	stateServerIP
+	stateServerCert
+	stateHostedMenderCredentials
+	statePolling
+	stateDone
+)
+
+const ( // Prompt constants
+	promptWizard = `Mender Client Setup
+===================
+
+Setting up the Mender client: the client runs on your device as a daemon, and will regularly poll the server to check for updates and report its inventory data.
+
+Get started by first configuring the device type and settings for communicating with the server:
+`
+	promptDone = "Starting Mender client…\n" + // NOTE: format
+		"Your device should show up at %s within the next few " +
+		"minutes.\nLog in to your Mender UI at %s to accept it.\n"
+
+	promptDeviceType   = "Device type (for example raspberrypi3-raspbian): "
+	promptConfirmDev   = "Confirm devicetype as %s? [Y/n] " // NOTE: format
+	promptHostedMender = "\nAre you connecting this device to Hosted Mender? [Y/n] "
+	promptCredentials  = "\nEnter your Hosted Mender credentials"
+	promptDemoMode     = "\nDemo mode uses short poll intervals and assumes the " +
+		"default demo server setup. (Recommended for testing.)\n" +
+		"Do you want to run the client in demo mode? [Y/n] "
+
+	promptServerIP   = "\nSet the IP of the Mender Server: [127.0.0.1] "
+	promptServerURL  = "\nSet the URL of the Mender Server: [https://docker.mender.io] "
+	promptServerCert = "\nSet the location of the certificate of the server, " +
+		"if any (path to file, for example /etc/mender/server.crt): "
+	promptUpdatePoll = "\nSet the update poll interval - the frequency with " +
+		"which the client will send an update check request to the " +
+		"server, in seconds: [1800] "
+	promptRetryPoll = "\nThe retry poll interval defines the number of " +
+		"seconds between each attempt the client tries to communicate " +
+		"with the server. Note that the client may attempt more often " +
+		"initially to enable rapid upgrades, but will gradually fall " +
+		"back to this value if the server is busy.\n" +
+		"Set the retry poll interval [300]: "
+	promptInventoryPoll = "Set the inventory poll interval - the frequency " +
+		"with which the client will send inventory data to the server, " +
+		"in seconds: [28800] "
+
+	// Response on invalid input
+	rspNoDevice      = "Device type cannot be blank: "
+	rspInvalidDevice = `The device type \"%s\" contains spaces or special
+Characters. Please try again: `
+	rspSelectYN     = "Please select Y or N: "
+	rspInvalidEmail = "\n\"%s\" does not appear to be a " + // NOTE: format
+		"valid email address.\nPlease enter a valid email address: "
+	rspHMLogin = "We couldn’t find a Hosted Mender account with those " +
+		"credentials.\nPlease try again: "
+	rspConnectionError = `There was a problem connecting to hosted.mender.io.
+Please check your device’s connection and try again.\n`
```

Comment:
This message (probably others!) does not format correctly, as the \n is not escaped.

```
There was a problem connecting to hosted.mender.io.
Please check your device’s connection and try again.\nEmail:
```